asllib/Typing.ml

```diff
@@ -3297,18 +3568,19 @@ module Annotate (C : ANNOTATE_CONFIG) : S = struct
             if extra_fields = [] then ty
             else
               match IMap.find_opt super genv.declared_types with
-              | Some { desc = T_Record fields; _ } ->
+              | Some ({ desc = T_Record fields; _ }, _) ->
                   T_Record (fields @ extra_fields) |> add_pos_from_st ty
-              | Some { desc = T_Exception fields; _ } ->
+              | Some ({ desc = T_Exception fields; _ }, _) ->
                   T_Exception (fields @ extra_fields) |> add_pos_from_st ty
               | Some _ -> conflict loc [ T_Record []; T_Exception [] ] ty
               | None -> undefined_identifier loc super
           and env = add_subtype name super env in
           (env, new_ty)
       (* AnnotateExtraFields) *)
     in
-    let t2 = annotate_type ~decl:true ~loc env1 t1 in
-    let env2 = add_type name t2 env1 in
+    let t2, ses_t = annotate_type ~decl:true ~loc env1 t1 in
```

Comment:
Don't we need to check that type definitions side effects are statically evaluable?